services/provider/server/storageclaim.go

```diff
@@ -98,19 +98,49 @@ func (s *storageClassClaimManager) Create(
 		},
 	})
 
-	err = s.client.Create(ctx, storageClassClaimObj)
+	claimBytes, _ := json.Marshal(storageClassClaimObj)
+	secretObj := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				ocsv1alpha1.StorageClassClaimAnnotation: fmt.Sprintf("%s/%s", storageClassClaimObj.Namespace, storageClassClaimObj.Name),
+			},
+			Name:      fmt.Sprintf("%s-claim", storageClassClaimObj.Name),
+			Namespace: s.namespace,
+			Labels:    storageClassClaimObj.ObjectMeta.Labels,
+		},
+		Data: map[string][]byte{
+			"StorageClassClaim": claimBytes,
+		},
+	}
+	secretObj.SetOwnerReferences([]metav1.OwnerReference{
```

Comment:
Just a reminder to remove the SetOwnerReferences call on the claim object (located above this code change)

Reply:
Done